modules/pubCommonIdSystem.js

```diff
@@ -8,20 +8,135 @@
 
+/**
+ * Store sharedid in either cookie or local storage
+ * @param {Object} config Need config.storage object to derive key, expiry time, and storage type.
+ * @param {string} value Shareid value to store
+ */
+
+function storeData(config, value) {
+  if (value) {
+    const key = config.storage.name + SHAREDID_SUFFIX;
+    if (config.storage.type === COOKIE) {
+      if (storage.cookiesAreEnabled()) {
+        const expiresStr = (new Date(Date.now() + (storage.expires * (60 * 60 * 24 * 1000)))).toUTCString();
+        storage.setCookie(key, value, expiresStr, 'LAX', COOKIE_DOMAIN);
+      }
+    } else if (config.storage.type === LOCAL_STORAGE) {
+      if (storage.hasLocalStorage()) {
+        storage.setDataInLocalStorage(key, value);
+      }
+    }
+  }
+}
+
+/**
+ * Read sharedid from cookie or local storage
+ * @param config Need config.storage to derive key and storage type
+ * @return {string}
+ */
+function readData(config) {
+  const key = config.storage.name + SHAREDID_SUFFIX;
+  if (config.storage.type === COOKIE) {
+    if (storage.cookiesAreEnabled()) { return storage.getCookie(key); }
+  } else if (config.storage.type === LOCAL_STORAGE) {
+    if (storage.hasLocalStorage()) {
+      return storage.getDataFromLocalStorage(key);
+    }
+  }
+}
+
+/**
+ * Delete sharedid from cookie or local storage
+ * @param config Need config.storage to derive key and storage type
+ */
+function delData(config) {
+  const key = config.storage.name + SHAREDID_SUFFIX;
+  if (config.storage.type === COOKIE) {
+    if (storage.cookiesAreEnabled()) {
+      storage.setCookie(key, '', EXPIRED_COOKIE_DATE);
+    }
+  } else if (config.storage.type === LOCAL_STORAGE) {
+    storage.removeDataFromLocalStorage(key);
+  }
+}
+
+/**
+ * setup success and error handler for sharedid callback thru ajax
+ * @param {string} pubcid Current pubcommon id
+ * @param {function} callback userId module callback.
+ * @param {Object} config Need config.storage to derive sharedid storage params
+ * @return {{success: success, error: error}}
+ */
+
+function handleResponse(pubcid, callback, config) {
+  return {
+    success: function (responseBody) {
+      if (responseBody) {
+        try {
+          let responseObj = JSON.parse(responseBody);
+          utils.logInfo('SharedId: Generated SharedId: ' + responseObj.sharedId);
+          if (responseObj.sharedId) {
+            if (responseObj.sharedId !== SHAREDID_OPT_OUT_VALUE) {
+              // Store sharedId locally
+              storeData(config, responseObj.sharedId);
+            } else {
+              // Delete local copy if the user has opted out
+              delData(config);
+            }
+          }
+          // Pass pubcid even though there is no change in order to trigger decode
+          callback(pubcid);
+        } catch (error) {
+          utils.logError(error);
+        }
+      }
+    },
+    error: function (statusText, responseBody) {
+      utils.logInfo('SharedId: failed to get id');
```

Comment:
This could be more clear about where it's coming from:

```suggestion
      utils.logInfo('PubCommonId: failed to get SharedId');
```